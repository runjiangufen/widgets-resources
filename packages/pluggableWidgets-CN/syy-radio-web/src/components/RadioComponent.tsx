import { createElement, ReactElement } from "react";
import { Radio, Space, RadioGroupProps, CheckboxOptionType } from "antd";

export interface RadioProps extends RadioGroupProps {
    direction?: string;
    options: CheckboxOptionType[];
}

export const RadioComponent = (props: RadioProps): ReactElement => {
    const { size, buttonStyle, disabled, name, optionType, onChange, direction, options, defaultValue } = props;

    const TAGS = optionType === "button" ? Radio.Button : Radio;
    return (
        <Radio.Group
            size={size}
            buttonStyle={buttonStyle}
            disabled={disabled}
            name={name}
            optionType={optionType}
            onChange={onChange}
            defaultValue={defaultValue}
        >
            {direction === "vertical" ? (
                <Space direction={direction}>
                    {options.map((item: any) => (
                        <TAGS key={item.value} value={item.value} disabled={item.disabled ? item.disabled : disabled}>
                            {item.label}
                        </TAGS>
                    ))}
                </Space>
            ) : (
                options.map((item: any) => (
                    <TAGS key={item.value} value={item.value} disabled={item.disabled ? item.disabled : disabled}>
                        {item.label}
                    </TAGS>
                ))
            )}
        </Radio.Group>
    );
};
