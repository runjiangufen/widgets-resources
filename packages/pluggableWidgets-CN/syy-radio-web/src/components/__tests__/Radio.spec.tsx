import { render, ShallowWrapper, shallow, mount } from "enzyme";
import { createElement } from "react";

import { RadioComponent, RadioProps } from "../RadioComponent";

describe("Radio", () => {
    const createRadio = (props: RadioProps): ShallowWrapper<any, any> => shallow(<RadioComponent {...props} />);

    let defaultRadioProps: RadioProps;

    beforeEach(() => {
        defaultRadioProps = {
            options: [
                { label: "Apple", value: "Apple" },
                { label: "Pear", value: "Pear" },
                { label: "Orange", value: "Orange" }
            ]
        };
    });

    it("renders as radioGroup", () => {
        const radioWrapper = createRadio(defaultRadioProps);
        expect(radioWrapper).toMatchSnapshot();
    });

    it("renders as radioGroup when direction is vertical", () => {
        defaultRadioProps.direction = "vertical";
        const radioWrapper = createRadio(defaultRadioProps);
        radioWrapper.find(".ant-space-vertical").exists();
    });

    it("renders as radioGroup when have defaultValue", () => {
        defaultRadioProps.defaultValue = "Pear";
        const radioWrapper = render(<RadioComponent {...defaultRadioProps} />);
        expect(radioWrapper).toMatchSnapshot();
    });

    it("renders as radioGroup when optionType is equal to button", () => {
        defaultRadioProps.optionType = "button";
        const radioWrapper = createRadio(defaultRadioProps);
        expect(radioWrapper).toMatchSnapshot();
    });

    it("renders as radioGroup when option is disabled", () => {
        defaultRadioProps.options[0].disabled = true;
        const radioWrapper = render(<RadioComponent {...defaultRadioProps} />);
        expect(radioWrapper).toMatchSnapshot();
    });

    it("renders as radioGroup when have option0 disabled", () => {
        defaultRadioProps.direction = "vertical";
        defaultRadioProps.options[0].disabled = true;
        const radioWrapper = render(<RadioComponent {...defaultRadioProps} />);
        expect(radioWrapper).toMatchSnapshot();
    });

    it("triggers onChange function with radiot change", () => {
        const onChange = jest.fn();
        defaultRadioProps.options = [{ label: "Apple", value: "Apple" }];
        const radioWrapper = mount(<RadioComponent {...defaultRadioProps} onChange={onChange} />);
        const inputWrapper = radioWrapper.find("input");
        inputWrapper.simulate("change");
        expect(onChange).toHaveBeenCalledTimes(1);
    });
});
