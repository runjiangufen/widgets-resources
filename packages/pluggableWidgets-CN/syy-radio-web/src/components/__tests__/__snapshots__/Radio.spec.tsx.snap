// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio renders as radioGroup 1`] = `
<Memo(ForwardRef)>
  <Radio
    key="Apple"
    value="Apple"
  >
    Apple
  </Radio>
  <Radio
    key="Pear"
    value="Pear"
  >
    Pear
  </Radio>
  <Radio
    key="Orange"
    value="Orange"
  >
    Orange
  </Radio>
</Memo(ForwardRef)>
`;

exports[`Radio renders as radioGroup when have defaultValue 1`] = `
<div
  class="ant-radio-group ant-radio-group-outline"
>
  <label
    class="ant-radio-wrapper"
  >
    <span
      class="ant-radio"
    >
      <input
        class="ant-radio-input"
        type="radio"
        value="Apple"
      />
      <span
        class="ant-radio-inner"
      />
    </span>
    <span>
      Apple
    </span>
  </label>
  <label
    class="ant-radio-wrapper ant-radio-wrapper-checked"
  >
    <span
      class="ant-radio ant-radio-checked"
    >
      <input
        checked=""
        class="ant-radio-input"
        type="radio"
        value="Pear"
      />
      <span
        class="ant-radio-inner"
      />
    </span>
    <span>
      Pear
    </span>
  </label>
  <label
    class="ant-radio-wrapper"
  >
    <span
      class="ant-radio"
    >
      <input
        class="ant-radio-input"
        type="radio"
        value="Orange"
      />
      <span
        class="ant-radio-inner"
      />
    </span>
    <span>
      Orange
    </span>
  </label>
</div>
`;

exports[`Radio renders as radioGroup when have option0 disabled 1`] = `
<div
  class="ant-radio-group ant-radio-group-outline"
>
  <div
    class="ant-space ant-space-vertical"
  >
    <div
      class="ant-space-item"
      style="margin-bottom:8px"
    >
      <label
        class="ant-radio-wrapper ant-radio-wrapper-disabled"
      >
        <span
          class="ant-radio ant-radio-disabled"
        >
          <input
            class="ant-radio-input"
            disabled=""
            type="radio"
            value="Apple"
          />
          <span
            class="ant-radio-inner"
          />
        </span>
        <span>
          Apple
        </span>
      </label>
    </div>
    <div
      class="ant-space-item"
      style="margin-bottom:8px"
    >
      <label
        class="ant-radio-wrapper"
      >
        <span
          class="ant-radio"
        >
          <input
            class="ant-radio-input"
            type="radio"
            value="Pear"
          />
          <span
            class="ant-radio-inner"
          />
        </span>
        <span>
          Pear
        </span>
      </label>
    </div>
    <div
      class="ant-space-item"
    >
      <label
        class="ant-radio-wrapper"
      >
        <span
          class="ant-radio"
        >
          <input
            class="ant-radio-input"
            type="radio"
            value="Orange"
          />
          <span
            class="ant-radio-inner"
          />
        </span>
        <span>
          Orange
        </span>
      </label>
    </div>
  </div>
</div>
`;

exports[`Radio renders as radioGroup when option is disabled 1`] = `
<div
  class="ant-radio-group ant-radio-group-outline"
>
  <label
    class="ant-radio-wrapper ant-radio-wrapper-disabled"
  >
    <span
      class="ant-radio ant-radio-disabled"
    >
      <input
        class="ant-radio-input"
        disabled=""
        type="radio"
        value="Apple"
      />
      <span
        class="ant-radio-inner"
      />
    </span>
    <span>
      Apple
    </span>
  </label>
  <label
    class="ant-radio-wrapper"
  >
    <span
      class="ant-radio"
    >
      <input
        class="ant-radio-input"
        type="radio"
        value="Pear"
      />
      <span
        class="ant-radio-inner"
      />
    </span>
    <span>
      Pear
    </span>
  </label>
  <label
    class="ant-radio-wrapper"
  >
    <span
      class="ant-radio"
    >
      <input
        class="ant-radio-input"
        type="radio"
        value="Orange"
      />
      <span
        class="ant-radio-inner"
      />
    </span>
    <span>
      Orange
    </span>
  </label>
</div>
`;

exports[`Radio renders as radioGroup when optionType is equal to button 1`] = `
<Memo(ForwardRef)
  optionType="button"
>
  <ForwardRef(RadioButton)
    key="Apple"
    value="Apple"
  >
    Apple
  </ForwardRef(RadioButton)>
  <ForwardRef(RadioButton)
    key="Pear"
    value="Pear"
  >
    Pear
  </ForwardRef(RadioButton)>
  <ForwardRef(RadioButton)
    key="Orange"
    value="Orange"
  >
    Orange
  </ForwardRef(RadioButton)>
</Memo(ForwardRef)>
`;
